source/base/Text.ooc

```diff
@@ -53,6 +53,30 @@ Text: cover {
 	operator != (other: String) -> Bool { !(this == other) }
 	operator != (other: This) -> Bool { !(this == other) }
 	operator + (other: This) -> This { This new(this _buffer + other _buffer) }
+	operator + (other: Int) -> This {
+		converted := other toString()
```

Comment:
Maybe it would be a good idea to write a short program that converts between `String` and `Text` (importantly, both literals and non-literals) that does not leak memory and that does not result in invalid reads/writes (which I'm sure a memtest will prove is the reason for the "weird characters" in your program).